### ISSUE TITLE
Add 'tabview' functionality?

### ISSUE THREAD

**Issue by pete-pjb** [kisvegabor/lvgl#3407]
Further to my recent forum support [here](https://forum.lvgl.io/t/anyone-know-how-to-rename-a-tabview-tab-after-its-been-created/9038?u=pete-pjb)  I would like to propose we add a function to the 'tabview' widget to provide a less hacky way to rename a tab on the 'tabview' after it is created.   Also the hacky method in the forum post will fall down if the tabs are vertical! I am really not a fan of hacky code anyway!

Can I suggest we introduce an API function prototype:

```c
void lv_tabview_rename_tab( lv_obj_t *obj, uint16_t tab_id, const char *new_name );
```

Implemented as follows:
```c
void lv_tabview_rename_tab( lv_obj_t *obj, uint32_t id, const char *new_name )
{
    LV_ASSERT_OBJ(obj, MY_CLASS);
    lv_tabview_t * tabview = (lv_tabview_t *)obj;

    if(id >= tabview->tab_cnt) return;
    if( tabview->tab_pos & LV_DIR_HOR ) id *= 2;

    lv_mem_free( tabview->map[id] );
    tabview->map[id] = lv_mem_alloc(strlen( new_name )+1 );
    strcpy( tabview->map[id], new_name );
    lv_obj_invalidate(obj);
}
```

This I hope will work well for the time being avoiding my nasty suggested hack!

@kisvegabor what do you think?  

If this is okay I can get this added to the master branch today.

Cheers Pete 🙂

**Comment by kisvegabor** [comment]
I agree to add this feature. v8 will be supported for 1 year after v8.3 is released so there is long time while people can make good use of it.

So please, send a PR if you have some time. :slightly_smiling_face:

**Comment by pete-pjb** [comment]
Hi @kisvegabor ,

I have committed the changes to the library and the docs [here.](https://github.com/lvgl/lvgl/commit/2c9695afb4ed6597ae54806c5eb2a287925343f8)

Then eventually sorted the format, sorry for the multiple pushes!  I use Windows and didn't have python working for the format script but I have now added python to my Windows environment and can now run the script.

PS is there anything else I should do to update the docs?  I have just added the stuff to tabview.md file so far.

Kind Regards,

Pete
